gtsfm/frontend/correspondence_generator/keypoint_aggregator/keypoint_aggregator_dedup.py

```diff
@@ -45,11 +51,16 @@ def append_unique_keypoints(
 
         for k, uv in enumerate(keypoints.coordinates):
             diff_norms = np.linalg.norm(per_image_kpt_coordinates[i] - uv, axis=1)
-            # TODO(johnwlambert,ayushbaid): test loosening threshold below to some epsilon.
-            is_identical = np.any(diff_norms == 0)
-            if len(per_image_kpt_coordinates[i]) > 0 and is_identical:
+            is_duplicate = np.any(diff_norms <= self.nms_merge_radius)
+            if len(per_image_kpt_coordinates[i]) > 0 and is_duplicate:
```

Comment:
have you tested with len(per_image_kpt_coordinates[i]) == 0? does the `<= ` work with empty arrrays? 
Also np.any should be false in that case, so you would not need that condition at all?